service/api_task.go

```diff
@@ -323,27 +323,12 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 
 // prepareForReprovision readies host for reprovisioning.
 func prepareForReprovision(ctx context.Context, env evergreen.Environment, settings *evergreen.Settings, h *host.Host, w http.ResponseWriter) error {
-	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
-	defer cancel()
-
-	if err := h.StopAgentMonitor(ctx, env); err != nil {
-		grip.Error(message.WrapError(err, message.Fields{
-			"message":       "problem stopping agent monitor",
-			"host_id":       h.Id,
-			"operation":     "next_task",
-			"revision":      evergreen.BuildRevision,
-			"agent":         evergreen.AgentVersion,
-			"current_agent": h.AgentRevision,
-		}))
-		return err
-	}
-
 	if err := h.MarkAsReprovisioning(); err != nil {
 		return errors.Wrap(err, "error marking host as ready for reprovisioning")
 	}
-	if h.NeedsReprovision != host.ReprovisionJasperRestart {
-		event.LogHostConvertingProvisioning(h.Id, h.Distro.BootstrapSettings.Method)
-	}
+
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
Enqueueing the job here is just an optimization to make it run sooner. Since it's called within the next task route, I wouldn't want the request to be blocked waiting for this to happen.

Reply:
Waiting for the job to be enqueued in Amboy (not that I really expect a job insertion to typically take a long time).